homeassistant/components/advantage_air/climate.py

```diff
@@ -168,6 +168,7 @@ def __init__(self, instance, ac_key, zone_key):
 
     async def async_added_to_hass(self):
         """When entity is added to hass."""
+        self._update_callback()
```

Comment:
Should we adjust the `async_add_entities` during setup to update all entities on adding them to has instead?

Reply:
Not sure if I am understanding, do you suggest I overload `async_add_entities` in `async_setup_entry` to include that call? I had originally tested putting the `self._update_callback()` call in to __init__ but that causes `self.async_write_ha_state()` to fail.

Reply:
No, the problem seems to be that the entity needs to be updated on adding, we don't need that call for that, we can update async_add_entites to handle that.

Reply:
Certain attributes only exist inside `self._update_callback()`, such as '_attr_hvac_mode', so its just a race to get that defined before its added to HA. For example I could manually define them in init too but that would cause duplicate code.

How would I go about updating async_add_entites to handle this? I cant see anything in the docs or a quick search of other integrations that modify async_add_entities.

Reply:
Not modify, pass in the second parameter as True

Reply:
Understood, that should work. Ill update and fix the tests.

Reply:
It's still having the same issue when setting update=True. Because `_update_callback` is only called when the data coordinator does an update **after** the entity has been added to hass, so the property doesn't exist.

The integration does an `await coordinator.async_config_entry_first_refresh()` before `async_setup_platforms`, making the update=True semi-redundant in function, but if it triggered the listener hook that would solve this problem. However I am unsure if asking for another update will actually trigger the coordinator listener (because it was already updated at setup), and that listener is only added when the entity is added to hass.

Reply:
You need to either split the "state parsing" out of the `_update_callback` to avoid calling `async_write_ha_state` or you use the properties instead of `_attr_` properties and extract the data out of the coordinator instead of inside `_update_callback`

Reply:
Using _attr_ (and the issue I am trying to fix) was introduce by @tkdrob in #52498, should I create a PR to partially roll that back so the integration is using `@property` again for non-static values.

Reply:
I have removed `_update_callback` and restored the properties.